core/server/master/src/main/java/alluxio/master/file/meta/UfsSyncPathCache.java

```diff
@@ -33,18 +36,30 @@
 public final class UfsSyncPathCache {
   private static final Logger LOG = LoggerFactory.getLogger(UfsSyncPathCache.class);
 
-  /** Number of paths to cache. */
-  private static final int MAX_PATHS =
-      Configuration.getInt(PropertyKey.MASTER_UFS_PATH_CACHE_CAPACITY);
-
   /** Cache of paths which have been synced. */
-  private final Cache<String, SyncTime> mCache;
+  public final Cache<String, SyncTime> mCache;
 
   /**
    * Creates a new instance of {@link UfsSyncPathCache}.
+   *
+   * TODO(jiacheng): One potential issue is those NOT_SYNCED entries are not evicted
+   *   So the memory will not be released if BOTH:
+   *   (1) Many forced sync entries; (2) No sync really happens
+   *   However that should be a really rare case?
    */
-  public UfsSyncPathCache() {
-    mCache = CacheBuilder.newBuilder().maximumSize(MAX_PATHS).build();
+  public UfsSyncPathCache(int maxPaths) {
+    mCache = CacheBuilder.newBuilder()
+        .maximumWeight(maxPaths)
```

Comment:
See https://github.com/Alluxio/alluxio/pull/16141/files#r969624549 for how this works now